src/GraphQL.AotCompilationSample.TypeFirst/Program.cs

```diff
@@ -0,0 +1,103 @@
+// See https://aka.ms/new-console-template for more information
+using GraphQL;
+using GraphQL.StarWars.TypeFirst;
+using GraphQL.StarWars.TypeFirst.Types;
+using GraphQL.Types;
+using GraphQL.Types.Relay.DataObjects;
+using Microsoft.Extensions.DependencyInjection;
+
+Console.WriteLine("Sample of AOT compilation of a GraphQL query on a type-first schema");
+Console.WriteLine();
+
+IServiceCollection serviceCollection = new ServiceCollection();
+// build services as usual; most methods are fully supported except:
+//   - AddClrTypeMappings
+//   - AddAutoClrMappings
+//   - AddAutoSchema
+serviceCollection.AddGraphQL(b => b
+    .AddSystemTextJson()
+    .AddAutoSchema<StarWarsQuery>(c => c.WithMutation<StarWarsMutation>())
+);
+
+#pragma warning disable IL2111 // Method with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
+// all types within GraphQL.StarWars.TypeFirst are rooted via TrimmerRootAssembly in the csproj so they will not be trimmed
+// for enumeration types, must also root these two types (for each enum in the schema)
+Preserve<GraphQLClrOutputTypeReference<Episodes>>();
+Preserve<EnumerationGraphType<Episodes>>();
+// for connection types, must also root the connection types used in the schema
+Preserve<Connection<IStarWarsCharacter>>();
+Preserve<Edge<IStarWarsCharacter>>();
+Preserve<PageInfo>();
+#pragma warning restore IL2111 // Method with parameters or return value with `DynamicallyAccessedMembersAttribute` is accessed via reflection. Trimmer can't guarantee availability of the requirements of the method.
+
+serviceCollection.AddSingleton<StarWarsData>();
+
+// must manually register the query and mutation types or AOT will trim their constructors
+// all other graph types' constructors are preserved via calls to Field<T>
+//serviceCollection.AddTransient<StarWarsQuery>();
+//serviceCollection.AddTransient<StarWarsMutation>();
```

Comment:
If "must" then why are 2 lines commented?

Reply:
I'll update the comment.  They are now rooted by the csproj